### PR TITLE
Convert scanpopup.ui to code, fix a few minor problems as side effects

### PR DESCRIPTION
**Pull request by shenlebantongying**
The net line change of scanpopup.cc is 0

Better layout, faster and simpler.

----

After the Change:
image

Super long:
image

The right size has a permanent white space, the reason is similar to browsers in which they also reserve a space for users to drag.

image

Side effects:

The original code contains a multiple Signal(clicked) → Slot(on_button_clicked) → Slot(View->doSomething).
https://github.com/xiaoyifang/goldendict/blob/3173a134936b9e8839c0ea1a0242c1496d4e7957/scanpopup.cc#L971-L974

They are simplified into Signal(clicked) → Slot(View→doSomething). The extra middle step is gone.

There used be 2 layers of QWidget and QFrame wrapped around scanpopup, and now they are gone.

### CODE REVIEW

**Comment by xiaoyifang** [comment]
> After the Change: image
> 
> Super long:

the original code is adjusting the height ,the screenshot is for the width ,they are not the same behaviour.

**Comment by shenlebantongying** [comment]
The height adjustment screenshot is this one, after the change, which is exactly the same as before but less hacky.

image

File: src/ui/scanpopup.cc [hunk 1]
```diff
@@ -764,17 +790,6 @@ bool ScanPopup::eventFilter( QObject * watched, QEvent * event )
...
-    {
-      // The UI looks ugly when group combobox is higher than translate line.
-      // Make the height of the combobox the same as the line edit's height.
-      // The fonts of these UI items should be kept in sync by applyWordsZoomLevel()
-      // so that text in the combobox is not clipped.
-      const QResizeEvent * const resizeEvent = static_cast< const QResizeEvent * >( event );
-      ui.groupList->setFixedHeight( resizeEvent->size().height() );
-      return false;
-    }
   }
```
Comment (xiaoyifang) [inline src/ui/scanpopup.cc:793]: I'd like this change to avoid use such hacks.

**Comment by xiaoyifang** [comment]
> The height adjustment screenshot is this one, after the change, which is exactly the same as before but less hacky.
> 
> image

does this mean ,  when user zoom the translationlin through the menu ,  the combox's height will not change

**Comment by xiaoyifang** [comment]
this PR is morely a rename ,  not see to much benefit.

**Comment by shenlebantongying** [comment]
> does this mean that when user zoom the translationline through the menu , the combox's height will not change

No, the zoom in the menu has never done anything to the ComboBox's size.

Only `Alt +-` can change GroupComboBox+TranslateLine's size. 

The code deleted is used to reacting to `QEvent::resize` which is widget's size changing (happens when the entire window resize or pressing "Alt +-").

**Comment by shenlebantongying** [comment]
I used code before change and remove/readd that section of code. It appears to me that it in fact does nothing in our codebase.

File: src/ui/scanpopup.cc (hunk 1 above)
Comment (xiaoyifang) [reply src/ui/scanpopup.cc:793]: this part of code can seperated into another PR.

**Comment by xiaoyifang** [comment]
> https://github.com/xiaoyifang/goldendict/blob/32304d4b152e578610a4779b8e355e3ff440ccbf/src/qtstyle/qt-style.css#L94-L96
> 

what's the result of this stylesheet, seems not a change worth refactoring the scanpopup.ui wholely.

**Comment by shenlebantongying** [comment]
> what's the result of this stylesheet
It is a thin line around the window:
image


The problem is that without this thin line, the scanpopup window cannot be easily distinguished. The problem is probably worse on Windows11 since the background on Windows is white by default.
image

**Comment by xiaoyifang** [comment]
> Without this thin line, the scanpopup window cannot be easily distinguished. The problem is probably worse on Windows11, where the background is white by default.

the result of frameless widgets

**Comment by xiaoyifang** [comment]
```
  setFrameShape(QFrame::Box);
//  setFrameShadow(QFrame::Raised);
  setLineWidth(1);
```

**Comment by shenlebantongying** [comment]
> the result of frameless widgets

Yes.

If you look closely to Windows' window, there is a 1px border. I think we should draw a thin line, too. The popup window is not distinguishable.

**Comment by shenlebantongying** [comment]
>   setFrameShape(QFrame::Box);
> //  setFrameShadow(QFrame::Raised);
>  setLineWidth(1);

I tried it before, the frame method has some problems with "pin mode"

image

**Comment by shenlebantongying** [comment]
The new code is simpler.

The ui file is
```
[a layer of QWidget [outframe  [translateline] [buttons]
    <Articleview appended in scanpopup.cc at run time>
    <Dictionarybar appended in scanpopup.cc> at run time] ]
```
The new code:

scanpopup_bar:
```
[[translate line] [buttons]]
```
Then assemble at mainwindow
```
[Layout [scanpopup_bar]
        [ArticleView]
        [DictionaryBar] ]
```

**Comment by xiaoyifang** [comment]
> > setFrameShape(QFrame::Box);
> > //  setFrameShadow(QFrame::Raised);
> > setLineWidth(1);
> 
> I tried it before, the frame method has some problems with "pin mode" → double border
> 
> image

maybe add a  shadow but no border

**Comment by shenlebantongying** [comment]
We can stick to ui file, but some adjustments are needed.

structure of .ui:
```
[ CentralWidget -> QWidget
   [ outframe -> QFrame
     [[ translateline ][ buttons ]]
     <Articleview not wirtten in .ui>
     <DictBar     not wirtten in .ui>
   ]
] 
```
Remove the frame & the wrapper widget from scanpopup.ui then rename it to scanpopup_bar.ui which contains only `[[ translateline ][ buttons ]]` part.


This looks like more reasonable.
```bash
# scanpopup_bar.ui

[[ translateline ][ buttons ]]

# scanpopup.cc

[frame [scanpopup_bar]
       [ArticleView]
       [DictionaryBar] ]
```

**Pull request closed without merging**